uk.co.compucorp.civicrm.hrleaveandabsences/api/v3/LeaveRequest.php

```diff
@@ -429,3 +429,198 @@ function civicrm_api3_leave_request_ismanagedby($params) {
 
   return $result;
 }
+
+/**
+ * LeaveRequest.addComment API spec
+ *
+ * @param $spec
+ */
+function _civicrm_api3_leave_request_addcomment_spec(&$spec) {
+  $spec['comment_id'] = [
+    'name' => 'comment_id',
+    'type' => CRM_Utils_Type::T_INT,
+    'title' => 'Comment ID',
+    'description' => 'The Comment ID',
+    'api.required' => 0
+  ];
+
+  $spec['leave_request_id'] = [
+    'name' => 'leave_request_id',
+    'type' => CRM_Utils_Type::T_INT,
+    'title' => 'Leave Request ID',
+    'description' => 'The ID of the Leave Request',
+    'api.required' => 1,
+  ];
+
+  $spec['text'] = [
+    'name' => 'text',
+    'type' => CRM_Utils_Type::T_STRING,
+    'title' => 'Text',
+    'description' => 'The comment text',
+    'api.required' => 1,
+  ];
+
+  $spec['contact_id'] = [
+    'name' => 'contact_id',
+    'type' => CRM_Utils_Type::T_INT,
+    'title' => 'Contact (Commenter)',
+    'description' => 'The contact who made the comment',
+    'api.required' => 1,
+    'FKClassName'  => 'CRM_Contact_DAO_Contact',
+    'FKApiName'    => 'Contact',
+  ];
+}
+
+/**
+ * LeaveRequest.addComment API
+ *
+ * Uses the API exposed by the Comment entity (Comment.create)
+ * to create comments related to a LeaveRequest.
+ *
+ * @param $params
+ *
+ * @return array
+ */
+function civicrm_api3_leave_request_addcomment($params) {
+  _civicrm_api3_leave_request_prepare_parameters_for_comment_payload($params);
+  $results =  civicrm_api3('Comment', 'create', $params);
+
+  if($results['count'] > 0) {
+    _civicrm_api3_leave_request_filter_comment_return_data($results['values']);
+  }
+
+  return $results;
+}
+
+/**
+ * LeaveRequest.getComment API spec
+ *
+ * @param $spec
+ */
+function _civicrm_api3_leave_request_getcomment_spec(&$spec) {
+  $spec['comment_id'] = [
+    'name' => 'comment_id',
+    'type' => CRM_Utils_Type::T_INT,
+    'title' => 'Comment ID',
+    'description' => 'The Comment ID',
+    'api.required' => 0
+  ];
+
+  $spec['leave_request_id'] = [
+    'name' => 'leave_request_id',
+    'type' => CRM_Utils_Type::T_INT,
+    'title' => 'Leave Request ID',
+    'description' => 'The ID of the Leave Request',
+    'api.required' => 0,
```

Comment:
I think this should be required. The reasoning is:
- In order to be able use the index on the Comments table, we need to pass both the entity_id and the entity_name
- I don't think there will be a case where we're going to need to load all the comments linked leave requests